Validation/SiTrackerPhase2V/plugins/Phase2ITValidateRecHit.cc

```diff
@@ -0,0 +1,439 @@
+    histoName.str("");
+    histoName << "Pull_X";
+    local_histos.pullX =
+        Phase2TkUtil::book1DFromPSet(config_.getParameter<edm::ParameterSet>("PullX"), histoName.str(), ibooker);
+    histoName.str("");
+    histoName << "Pull_Y";
+    local_histos.pullY =
+        Phase2TkUtil::book1DFromPSet(config_.getParameter<edm::ParameterSet>("PullY"), histoName.str(), ibooker);
+    histoName.str("");
+    histoName << "Delta_X_vs_Eta";
+    local_histos.deltaX_eta = Phase2TkUtil::bookProfile1DFromPSet(
+        config_.getParameter<edm::ParameterSet>("DeltaX_eta"), histoName.str(), ibooker);
+    histoName.str("");
+    histoName << "Delta_Y_vs_Eta";
+    local_histos.deltaY_eta = Phase2TkUtil::bookProfile1DFromPSet(
+        config_.getParameter<edm::ParameterSet>("DeltaX_eta"), histoName.str(), ibooker);
+    histoName.str("");
+    histoName << "Pull_X_vs_Eta";
+    local_histos.pullX_eta = Phase2TkUtil::bookProfile1DFromPSet(
+        config_.getParameter<edm::ParameterSet>("PullX_eta"), histoName.str(), ibooker);
+    histoName.str("");
+    histoName << "Pull_Y_vs_Eta";
+    local_histos.pullY_eta = Phase2TkUtil::bookProfile1DFromPSet(
+        config_.getParameter<edm::ParameterSet>("PullY_eta"), histoName.str(), ibooker);
+    ibooker.setCurrentFolder(subdir + "/" + key + "/PrimarySimHits");
+    //all histos for Primary particles
+    histoName.str("");
+    histoName << "Number_RecHits_matched_PrimarySimTrack";
+    local_histos.numberRecHitsprimary = Phase2TkUtil::book1DFromPSet(
+        config_.getParameter<edm::ParameterSet>("nRecHits_primary"), histoName.str(), ibooker);
+    histoName.str("");
+    histoName << "Delta_X_SimHitPrimary";
+    local_histos.deltaX_primary = Phase2TkUtil::book1DFromPSet(
+        config_.getParameter<edm::ParameterSet>("DeltaX_primary"), histoName.str(), ibooker);
+    histoName.str("");
+    histoName << "Delta_Y_SimHitPrimary";
+    local_histos.deltaY_primary = Phase2TkUtil::book1DFromPSet(
+        config_.getParameter<edm::ParameterSet>("DeltaY_primary"), histoName.str(), ibooker);
+    histoName.str("");
+    histoName << "Pull_X_SimHitPrimary";
+    local_histos.pullX_primary = Phase2TkUtil::book1DFromPSet(
+        config_.getParameter<edm::ParameterSet>("PullX_primary"), histoName.str(), ibooker);
+    histoName.str("");
+    histoName << "Pull_Y_SimHitPrimary";
+    local_histos.pullY_primary = Phase2TkUtil::book1DFromPSet(
+        config_.getParameter<edm::ParameterSet>("PullY_primary"), histoName.str(), ibooker);
+    layerMEs_.insert(std::make_pair(key, local_histos));
+  }
+}
+void Phase2ITValidateRecHit::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+  // rechitValidIT
+  edm::ParameterSetDescription desc;
+  {
+    edm::ParameterSetDescription psd0;
+    psd0.add<std::string>("name", "Delta_X");
+    psd0.add<std::string>("title", "Delta_X;RecHit resolution X dimensiom");
```

Comment:
@sroychow 
since you'be been asked to revise the code, please fix the typo.

```suggestion
    psd0.add<std::string>("title", "Delta_X;RecHit resolution X dimension");
```

Reply:
done in both cases.